src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -216,8 +215,20 @@ private enum Flags : byte
             return (SynthesizedBackingFieldSymbol?)_lazyBackingFieldSymbol;
         }
 
+        private bool CanHaveBackingField()
+        {
+            return !(_containingType.IsInterface && !IsStatic) && !IsAbstract && !IsExtern && !IsIndexer;
+        }
+
         internal SynthesizedBackingFieldSymbol? GetOrCreateBackingFieldForFieldKeyword()
         {
+            Debug.Assert(!IsIndexer);
+            if (!CanHaveBackingField())
```

Comment:
>if (!CanHaveBackingField()) [](http://example.com/codeflow?start=12&length=27)

The logic here doesn't feel right to me. I think interpretation of a `field` keyword should depend only on what is in scope, not on whether containing type is interface or on the property kind. Instead, we probably should report an error that "Interfaces cannot contain instance fields", etc. Basically, report an error that would be reported for an explicit field declaration. Suppressing creation of a symbol in error scenarios doesn't look useful on its own, for example we always create a symbol when there is an initializer.